functional_tests/steps/bundles.py

```diff
@@ -121,3 +128,267 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
+    context.page.get_by_role("link", name="Add bundle").click()
+    context.page.get_by_role("textbox", name="Name*").click()
+    context.page.get_by_role("textbox", name="Name*").fill(bundle_name)
+    context.page.get_by_role("button", name="Save as draft").click()
+    context.page.get_by_role("link", name="Dashboard").click()
+    expect(context.page.get_by_text("Latest active bundles")).to_be_visible()
+    expect(context.page.get_by_role("link", name=bundle_name)).to_be_visible()
+
+
+@then("the user cannot create a bundle")
+def the_user_cannot_add_bundles(context: Context) -> None:
+    expect(context.page.get_by_role("link", name="Add bundle")).not_to_be_visible()
+    expect(context.page.get_by_text("Bundles ready for preview")).to_be_visible()
+    expect(context.page.get_by_text("There are currently no")).to_be_visible()
+
+
+@step("the user can edit a bundle")
+def can_edit_bundle(context: Context) -> None:
+    expect(context.page.locator("#latest-bundles-heading")).to_contain_text("Latest active bundles")
+    expect(context.page.locator("#latest-bundles-content")).to_contain_text(context.bundles[0].name)
+    context.page.get_by_role("row", name=context.bundles[0].name + " Actions").get_by_label("Actions").click()
+    context.page.get_by_role("link", name="Edit", exact=True).click()
+
+    add_reslease_calendar_in_edit(context)
+    context.page.get_by_role("button", name="Save").click()
+    add_article_page_in_edit(context)
+    context.page.get_by_role("button", name="Save").click()
+    add_preview_team_in_edit(context)
+    context.page.get_by_role("button", name="Save").click()
+
+    context.page.get_by_role("button", name="More actions").click()
+    context.page.get_by_role("button", name="Save to preview").click()
+
+
+def add_preview_team_in_edit(context: Context) -> None:
+    # add preview team
+    if hasattr(context, "teams"):
+        expect(context.page.locator("#panel-preview_teams-heading")).to_contain_text("Preview teams")
+        context.page.get_by_role("button", name="Add preview team").click()
+        expect(context.page.get_by_role("textbox", name="Search term")).to_be_visible()
+        context.page.get_by_role("textbox", name="Search term").click()
+        context.page.get_by_role("textbox", name="Search term").fill(context.teams[0].name)
+        expect(context.page.get_by_role("checkbox", name=context.teams[0].name)).to_be_visible()
+        context.page.get_by_role("checkbox", name=context.teams[0].name).check()
+        context.page.wait_for_timeout(100)
+        expect(context.page.get_by_role("button", name="Confirm selection")).to_be_visible()
+        context.page.get_by_role("button", name="Confirm selection").click()
+
+
+def add_article_page_in_edit(context: Context) -> None:
+    # add Article
+    if hasattr(context, "statistical_article_pages"):
+        expect(context.page.get_by_role("heading", name="Bundled pages").locator("span")).to_be_visible()
```

Comment:
The `expect` calls here are redundant, you are performing an action, not making a check

Reply:
I think you mean "unit tests" instead of "functional tests"? As these are functional tests.
These `expect` calls are not needed because your actions will fail if the element doesn't exist, so the calls are redundant. If you look at other code you'll see we don't follow this pattern.